src/app/contexts/UserContext/index.test.tsx

```diff
@@ -61,4 +64,71 @@ describe('UserContext', () => {
       {},
     );
   });
+  describe('ckns_mvt cookie', () => {
+    const cookieSetterSpy = jest.spyOn(Cookie, 'set');
+    const cookieGetterSpy = jest.spyOn(Cookie, 'get');
+    const isOperaProxySpy = jest.spyOn(isOperaProxy, 'default');
+    const onClientSpy = jest.spyOn(onClient, 'default');
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('should call cookie logic when not opera mini and is on client', () => {
+      onClientSpy.mockImplementationOnce(() => true as unknown as Location);
+      isOperaProxySpy.mockImplementationOnce(() => false);
+
+      render(<DummyComponentWithContext />);
+
+      expect(cookieGetterSpy).toHaveBeenCalled();
+    });
+
+    it('should not call cookie logic when on opera mini and is on client', () => {
+      onClientSpy.mockImplementationOnce(() => true as unknown as Location);
+      isOperaProxySpy.mockImplementationOnce(() => true);
+
+      render(<DummyComponentWithContext />);
+
+      expect(cookieGetterSpy).not.toHaveBeenCalled();
+    });
+
+    it('should not call cookie logic when not on opera mini and is not on client', () => {
+      onClientSpy.mockImplementationOnce(() => false);
+      isOperaProxySpy.mockImplementationOnce(() => false);
+
+      render(<DummyComponentWithContext />);
+
+      expect(cookieGetterSpy).not.toHaveBeenCalled();
+    });
+
+    it('should not set cookie when ckns_mvt cookie exists', () => {
```

Comment:
```suggestion
    it('should not set cookie when ckns_mvt cookie already exists', () => {
```

Reply:
I am not seeing any references to the `ckns_mvt` in this test - do we need to have an explicit mock for this (rather than just foo?) e.g. we did a form of cookie mocking here: https://github.com/bbc/simorgh/blob/52633ffc5de47ad11912976f55eafc9152012301/src/app/lib/utilities/setCookie/index.test.jsx#L17 wonder if this might be helpful i.e. allow us to pretend that  `ckns_mvt` is already set?

Reply:
Yeah I took this approach as we're only really bothered if Cookie.set is called, but I agree it does make it more clear is ckns_mvt is set properly. I've updated the test to reflect this, clearing the setter mock before we render the user context so it doesn't mess with the expect

Reply:
I'm now using the setCookie() function here to do this now, as it has functionality to set the cookie properly on safari, so I've tried to update the test locally to reflect this, however now that I'm using setCookie(), the Cookie.get() in the code, can't find the cookie set with setCookie(), but the spy on Cookie.Set() has been called, is there something I need to do so that the actual code can recognise that the cookie has been set?

```ts
it('should not set cookie when ckns_mvt cookie already exists', () => {
      onClientSpy.mockImplementationOnce(() => true as unknown as Location);
      isOperaProxySpy.mockImplementationOnce(() => false);
      setCookie({ name: 'ckns_mvt', value: 'foo' });
      // expect(cookieSetterSpy).not.toHaveBeenCalled();
      cookieSetterSpy.mockClear();

      render(<DummyComponentWithContext />);

      expect(cookieSetterSpy).not.toHaveBeenCalled();
      expect(cookieGetterSpy).toHaveReturnedWith('foo');
    });
```

Reply:
I am wondering if the failing unit tests are to do with mocking...?

Reply:
The failing Next.js tests have nothing to do with this as far as I'm aware, this is an [issue with the setCookie utility](https://github.com/bbc/simorgh/blob/24674f13cac2a9036cab21bdf7bde4457da9031a/src/app/lib/utilities/setCookie/index.jsx#L36), this change seems to fix it:

```
  return Cookie.set(name, value, {
    expires,
    domain: document?.domain ? getCookieDomain(document.domain) : undefined,
    sameSite: sameSiteValues[sameSite],
    ...(isHttps && { secure: true }),
  });
```